### DISCUSSION TITLE
[DISCUSS] Support Remote Write to Prometheus for Leveraging Prometheus Ecosystem Benefits

### DISCUSSION THREAD

**Discussion by LeoQuote** [apache/skywalking#12104] [SWIP]
## Motivation
SkyWalking provides a wide range of metrics internally and already offers a prom-compatible URL for querying metrics. However, querying metrics is limited to display purposes only and cannot be used for setting alerts, leveraging Prometheus ecosystem features such as alertmanager, inhibit rules, and advanced query statements using promql syntax. By enabling the use of remote write to push metrics to the Prometheus ecosystem, integration between these two products can be enhanced, allowing users to utilize existing Prometheus operators for setting alerts and more.

## Architecture Graph
N/A

## Proposed Changes
- Implement the capability to push each metric or metrics within a whitelist to Prometheus through [remote write](https://prometheus.io/docs/concepts/remote_write_spec/).
- Expose a [federate API](https://prometheus.io/docs/prometheus/latest/federation/#configuring-federation) for Prometheus to freely fetch metrics.

## Imported Dependencies libs and their licenses
N/A

## Compatibility
N/A

## General usage docs
N/A

## Related
https://github.com/apache/skywalking/discussions/11666

**Comment by wu-sheng** [comment]
If you want to write to someplace else, pick the exporter of metrics, https://skywalking.apache.org/docs/main/latest/en/setup/backend/exporter/#metrics-grpc-exporter

And another component could run separately, or as another exporter implementation.

Could you be a more clear about your proposal, what is remote write? SkyWalking doesn't have this concept.

**Reply by LeoQuote** [reply]
Thanks for your quick reply, I've updated some links for remote write and federate, hope that would help, I'll look into the document you provided.

**Reply by wu-sheng** [reply]
I know what is Prometheus remote write. But you write the proposal in SkyWalking, the key is how you extend SkyWalking, not revertedly, right?

**Reply by LeoQuote** [reply]
Certainly, but I dont have much knowledge about skywalking, I think it needs more discussion. I think what you provided could be a good way and even better way than implementing this inside skywalking, I'll look into that, thanks for your help!

**Reply by wu-sheng** [reply]
Then, why do you send SWIP to SkyWalking? Proposal is for users to extend to wider ecosystems, and it should be real case oriented.

**Comment by LeoQuote** [comment]
Closing as skywalking has already provide tools to achive that, I'll share if there's more progress. Thanks @wu-sheng for the help!
